src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/Thread.NativeAot.cs

```diff
@@ -198,8 +198,22 @@ public int ManagedThreadId
             get => _managedThreadId.Id;
         }
 
-        // TODO: Inform the debugger and the profiler
-        // private void ThreadNameChanged(string? value) {}
+        // TODO: Support non-current thread
+        private unsafe partial void ThreadNameChanged(string? value)
+        {
+            if (Thread.CurrentThread != this)
+            {
+                return;
+            }
+            if (value == null)
+            {
+                return;
+            }
+            fixed (char* name = value)
+            {
+                RuntimeImports.RhSetCurrentThreadName(name);
+            }
```

Comment:
```suggestion
            RuntimeImports.RhSetCurrentThreadName(value);
```
You can leave it to the marshaller to do the pinning